lib/galaxy/datatypes/interval.py

```diff
@@ -313,7 +316,7 @@ def sniff( self, filename ):
         >>> Interval().sniff( fname )
         True
         """
-        headers = get_headers( filename, '\t', comment_designator='#' )
+        headers = iter_headers( filename, '\t', comment_designator='#' )
```

Comment:
I'd move the `iter_headers()` call inside the `try` (i.e. `for hdr in iter_headers( filename, '\t', comment_designator='#' )` ) and remove this line. Similarly below.